### PR TITLE
Adds date type question to the formview

### LINKED ISSUE #178: Add .date question type
**Description**
This question type asks the user to select a date (e.g. birthday)

**Acceptance Criteria**
- Use the same date picker as the one in catalog details
- Default date is today
- We will require backend to send us min and max date (e.g. user cannot select a date in the future for birth date) - NOT IMPLEMENTED YET
- If there are multiple date questions, only one date picker can be open at a given time

**Error Messages**
Any error messages that should be expected for invalid usage.

**UX Requirements**
Mockups, wireframes or any notes pertaining to the UX.

**Tech Requirements**
Design patterns, naming conventions, or any other considerations for the developer.

**QA Requirements**
What the QA should be testing for when evaluating this feature.

### PR DESCRIPTION
**Pull request by dorothyfu**
**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #178 

**Related issues that should be LINKED to this PR:**
* Connects #

### CODE REVIEW

File: traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift [hunk 1]
```diff
@@ -194,6 +238,64 @@ extension FormView: UICollectionViewDelegateFormLayout {
             fatalError("Unknown suppplementary element kind: \(kind)")
         }
     }
+
+    public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let _ = collectionView.cellForItem(at: indexPath) as? FormValueDisplayInputCell else {
+            return
+        }
+
+        UIView.setAnimationsEnabled(false)
```
Comment (ata-n) [inline traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: you shouldnt disable animations!...the goal was to fix the animations not disable them

File: traveler-swift-core/TravelerKit/Extensions/DateFormatter+PassengerKit.swift [hunk 2]
```diff
@@ -35,6 +35,12 @@ extension DateFormatter {
         return formatter
     }()
 
+    public static var shortFormatter: DateFormatter = {
```
Comment (ata-n) [inline traveler-swift-core/TravelerKit/Extensions/DateFormatter+PassengerKit.swift]: call this something else please..short format is `yyyy/mm/dd`

File: traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift (hunk 1 above)
Comment (ata-n) [reply traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: As I recall the issue was only around transparency...what needs to happen is the background on the cells need to be a solid color...thats it I think

File: traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormLayout.swift [hunk 3]
```diff
@@ -121,4 +121,17 @@ class FormLayout: UICollectionViewLayout {
...
+
+    override func initialLayoutAttributesForAppearingItem(at itemIndexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        let attributes = super.initialLayoutAttributesForAppearingItem(at: itemIndexPath)
+        attributes?.alpha = 1.0
+        return attributes
+    }
+
+    override func finalLayoutAttributesForDisappearingItem(at itemIndexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        let attributes = super.finalLayoutAttributesForDisappearingItem(at: itemIndexPath)
+        attributes?.transform = attributes!.transform.scaledBy(x: 0, y: 0.1)
```
Comment (Skrinikov) [inline traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormLayout.swift]: Why are you scaling it by 0.1?

File: traveler-swift-core/TravelerKit/Extensions/DateFormatter+PassengerKit.swift [hunk 4]
```diff
@@ -35,6 +35,12 @@ extension DateFormatter {
         return formatter
     }()
 
+    public static var monthTextFormatter: DateFormatter = {
```
Comment (Skrinikov) [inline traveler-swift-core/TravelerKit/Extensions/DateFormatter+PassengerKit.swift]: Can the name be something more descriptive?

File: traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormLayout.swift (hunk 3 above)
Comment (dorothyfu) [reply traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormLayout.swift]: To make the animation smoother but not obvious

File: traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift [hunk 5]
```diff
@@ -108,11 +124,24 @@ extension FormView: UICollectionViewDataSource {
...
     }
 
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let descriptor = dataSource!.formView(self, inputDescriptorForFieldAt: indexPath)
+        guard indexPath != datePickerIndexPath else {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: datePickerCellIdentifier, for: indexPath) as! FormDateInputCell
+            let valueCellIndexPath = IndexPath(item: indexPath.row - 1, section: indexPath.section)
+            cell.delegate = self
+            cell.datePicker.minimumDate = nil // TODO
+            cell.datePicker.maximumDate = nil // TODO
```
Comment (GeorgiosP) [inline traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: do the min and max date values just need to be added or is this something more involved?

Comment (dorothyfu) [reply traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: we have to wait for the API to implement it then add it to our models

Comment (GeorgiosP) [reply traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: ok sweet mind adding that to the comment 👍

Comment (dorothyfu) [reply traveler-ios-ui/TravelerKitUI/UIUtilities/FormView/FormView.swift]: done 👍

**Pull request merged**